### PR TITLE
Fix NuttX build after dev/socket merge

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: .github/workflows/compilation_on_nuttx.yml [hunk 1]
```diff
@@ -65,6 +65,7 @@ jobs:
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_FAST=y\\n",
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_CLASSIC=y\\n",
           "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_LIBC_BUILTIN=y\\n",
+          "CONFIG_INTERPRETERS_WAMR=y\\nCONFIG_INTERPRETERS_WAMR_AOT=y\\nCONFIG_INTERPRETERS_WAMR_LIBC_WASI=y\\nCONFIG_NET=y\\nCONFIG_NET_SOLINGER=y\\nCONFIG_NET_TCP=y\\nCONFIG_NET_TCP_WRITE_BUFFERS=y\\nCONFIG_NET_SOCKOPTS=y\\n",
```
Comment (loganek) [inline .github/workflows/compilation_on_nuttx.yml:68]: Could you clarify how the `CONFIG_NET*` parameters are used? I don't see any reference to them in the source code.

Comment (yamt) [reply .github/workflows/compilation_on_nuttx.yml:68]: libc-wasi now requires setsockopt etc and thus requires these nuttx options enabled.

Comment (loganek) [reply .github/workflows/compilation_on_nuttx.yml:68]: Could you point me to a code where those parameters are used? Thank you

Comment (yamt) [reply .github/workflows/compilation_on_nuttx.yml:68]: they are used in nuttx repo. https://github.com/apache/incubator-nuttx

**Comment by yamt** [comment]
i dropped the ci bits for now as it was more complex than i expected.
i will retry sooner or later. (probably in a separate PR)

**Review by loganek — APPROVED** [review]

**Pull request merged**